src/app/features/notifications/notifications.component.ts

```diff
@@ -94,18 +85,51 @@ export class NotificationsComponent implements OnInit {
   private filterUserNotifications(
     notifications: NotificationActivity[],
     user: string,
-  ): NotificationActivity[] {
-    return notifications.filter((notification) => notification.sentBy === user);
+  ) {
+    this.allNotifications = notifications.filter(
+      (notification) => notification.sentBy === user,
+    );
+    this.unreadNotifications = notifications.filter(
+      (notification) =>
+        notification.sentBy === user && notification.readStatus === false,
+    );
   }
 
   /**
    * Counts unread notifications from the list.
    * @param notifications - The list of notifications.
    */
-  private countUnreadNotifications(
-    notifications: NotificationActivity[],
-  ): number {
-    return notifications.filter((notification) => !notification.readStatus)
-      .length;
+  private countUnreadNotifications(notifications: NotificationActivity[]) {
+    return notifications.filter(
+      (notification) => notification.readStatus === false,
+    ).length;
+  }
+
+  markAllRead($event: Event) {
+    $event.stopPropagation();
+    Logging.log("All notifications marked as read");
+  }
+
+  markAsRead(notification: NotificationActivity): void {
+    // Need to add/update this logic to mark as read the notification from the CouchDB
+    notification.readStatus = true;
+    this.hasUnreadNotificationCount = this.countUnreadNotifications(
+      this.allNotifications,
+    );
+    Logging.log("Notification marked as read");
+  }
```

Comment:
This function is for when the user mark as read in single notification.

Reply:
I think you can simple replace the Logging with a call to EntityMapperService saving the updated notification entity?
`await this.entityMapper.save(notification);`